plugins/org.jkiss.dbeaver.ext.dm/src/org/jkiss/dbeaver/ext/dm/model/DMMetaModel.java

```diff
@@ -0,0 +1,157 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jkiss.dbeaver.ext.dm.model;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.ext.generic.model.*;
+import org.jkiss.dbeaver.ext.generic.model.meta.GenericMetaModel;
+import org.jkiss.dbeaver.model.DBPDataSourceContainer;
+import org.jkiss.dbeaver.model.DBUtils;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCPreparedStatement;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCSession;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCStatement;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.utils.CommonUtils;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Shengkai Bai
+ */
+public class DMMetaModel extends GenericMetaModel {
+
+    public DMMetaModel() {
+        super();
+    }
+
+    @Override
+    public GenericDataSource createDataSourceImpl(DBRProgressMonitor monitor, DBPDataSourceContainer container) throws DBException {
+        return new DMDataSource(monitor, container, this);
+    }
+
+    @Override
+    public boolean supportsSequences(GenericDataSource dataSource) {
+        return true;
+    }
+
+    @Override
+    public JDBCStatement prepareSequencesLoadStatement(JDBCSession session, GenericStructContainer container) throws SQLException {
+        JDBCPreparedStatement dbStat = session.prepareStatement(
+                "SELECT SEQ_OBJ_INNER.NAME,SEQ_OBJ_INNER.INFO1,SEQ_OBJ_INNER.INFO3,SEQ_OBJ_INNER.INFO4 " +
+                        "FROM " +
+                        "SYSOBJECTS SEQ_OBJ_INNER, " +
+                        "SYSOBJECTS SCH_OBJ_INNER " +
+                        "WHERE " +
+                        "SEQ_OBJ_INNER.SUBTYPE$ = 'SEQ' " +
+                        "AND SCH_OBJ_INNER.ID = SEQ_OBJ_INNER.SCHID " +
+                        "AND SCH_OBJ_INNER.name = ?");
+        dbStat.setString(1, container.getName());
+        return dbStat;
+    }
+
+    @Override
+    public GenericSequence createSequenceImpl(JDBCSession session, GenericStructContainer container, JDBCResultSet dbResult) throws DBException {
+        String name = JDBCUtils.safeGetString(dbResult, 1);
+        if (CommonUtils.isEmpty(name)) {
+            return null;
+        }
+        Number lastValue = JDBCUtils.safeGetBigDecimal(dbResult, 3);
+        Number incrementBy = JDBCUtils.safeGetBigDecimal(dbResult, 4);
+        return new GenericSequence(container, name, "", lastValue, null, null, incrementBy);
+    }
+
+    @Override
+    public boolean supportsDatabaseTriggers(GenericDataSource dataSource) {
```

Comment:
Does DaMeng support database triggers?
Because the super method of prepareContainerTriggersLoadStatement will return
throw new SQLFeatureNotSupportedException();
If container triggers are not supported, please remove this method.